starfish/core/experiment/experiment.py

```diff
@@ -105,105 +96,74 @@ def name(self) -> str:
     def image_types(self) -> Set[str]:
         return set(self._images.keys())
 
-    def show_aligned_image_groups(self) -> None:
-        """
-        Describe the aligned subgroups for each Tileset in this FOV
-
-        Examples
-        --------
-        >>> fov.show_aligned_image_groups()
-        {'nuclei': ' Group 0:  <starfish.ImageStack r={0}, ch={0}, z={0}, (y, x)=(190,270)>',
-        'primary': ' Group 0:  <starfish.ImageStack r={0, 1, 2, 3, 4, 5}, ch={0, 1, '
-        '2}, z={0}, (y, x)=(190, 270)>'}
-
-        The example describes a FieldOfView with two Tilesets (primary and nuclei), because
-        all images have the same (x, y) coordinates, each Tileset has a single aligned subgroup:
-        Group 0.
-        """
-        all_groups = dict()
-        for name, groups in self.aligned_coordinate_groups.items():
-            y_size = self._images[name].default_tile_shape[0]
-            x_size = self._images[name].default_tile_shape[1]
-            info = '\n'.join(
-                f" Group {k}: "
-                f" <starfish.ImageStack "
-                f"r={v._permitted_rounds if v._permitted_rounds else 1}, "
-                f"ch={v._permitted_chs if v._permitted_chs else 1}, "
-                f"z={v._permitted_zplanes if v._permitted_zplanes else 1}, "
-                f"(y, x)={y_size, x_size}>"
-                for k, v in enumerate(groups)
-            )
-            all_groups[name] = f'{info}'
-        pprint.pprint(all_groups)
-
-    def iterate_image_type(self, image_type: str) -> Iterator[ImageStack]:
-        """
-        Iterate through the aligned subgroups of the given image type.
-        (ex. primary)
-
-        Parameters
-        ----------
-        image_type : str
-            The name of the image type to iterate through.
-
-        Returns
-        --------
-        Iterator
-            An iterator for the aligned subgroups of the image type
-
-        """
-        for aligned_group, _ in enumerate(self.aligned_coordinate_groups[image_type]):
-            yield self.get_image(item=image_type, aligned_group=aligned_group)
-
-    def get_image(self, item: str, aligned_group: int = 0,
-                  rounds: Optional[Collection[int]] = None,
-                  chs: Optional[Collection[int]] = None,
-                  zplanes: Optional[Collection[int]] = None,
-                  x: Optional[Union[int, slice]] = None,
-                  y: Optional[Union[int, slice]] = None,
-                  ) -> ImageStack:
+    def get_images(self, item: str,
```

Comment:
I would prefer that `get_images` always returns an iterator.  We can leave `get_image` in the codebase, and consider deprecating it (a la `@deprecated`, or whatever similar annotation exists for python).

Reply:
I think this was something @ambrosejcarr specifically asked for? I will say it's kinda nice when you know your params are going to just return one imagestack like this allen dataset where I use get_images(rounds=[1]) and you don't have to then do a next(imagesatcks)

Reply:
It's probably convenient when you know exactly the nature of your data, and you just want to lazy-mode it.  In practice, I think this is likely going to be a lot of:

```
x = get_images(...)
if isinstance(x, ImageStack):
    do something
else:
    for img in x:
        do something
```

@ambrosejcarr you want to make the case for this? :)

Reply:
I think the UX I'd like is: 

`get_image() -> ImageStack`
If this 
`get_images() -> Iterator[ImageStack]`

Reply:
This might be a good place to get a consult from @kne42 as well.

Reply:
I'm cool with what you suggested.

Reply:
Yeah it should definitely always return an iterator for exactly the reason @ttung presented in his example.

If a user expects there to only be one `ImageStack`, they can unpack it like so:
```python
stack, = get_images(rounds=[1])
```
although I don't think this will be very intuitive to users unless you explicitly document it :)

I do think what @ambrosejcarr proposed with `get_image` and `get_images` is a good middle-ground in terms of convenience and intuitiveness.